daemon/cmd/daemon_main.go

```diff
@@ -1258,12 +1259,30 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
-	if option.Config.EnableIPSec && option.Config.Tunnel == option.TunnelDisabled && option.Config.EncryptInterface == "" {
-		link, err := linuxdatapath.NodeDeviceNameWithDefaultRoute()
-		if err != nil {
-			log.WithError(err).Fatal("Ipsec default interface lookup failed, consider \"encrypt-interface\" to manually configure interface.")
+	if option.Config.EnableIPSec && option.Config.Tunnel == option.TunnelDisabled && len(option.Config.EncryptInterface) == 0 {
+		// IPAMENI mode supports multiple network facing interfaces that
+		// will all need Encrypt logic applied in order to decrypt any
+		// received encrypted packets. This logic will attach to all
+		// !veth devices.
+		if option.Config.IPAM == ipamOption.IPAMENI {
+			interfaces := option.Config.EncryptInterface
+
+			if links, err := netlink.LinkList(); err == nil {
+				for _, link := range links {
+					use, err := ethtool.IsNotVirtualDriver(link.Attrs().Name)
```

Comment:
Looks like this only excludes `veth`* devices? How about `cilium_host` interface?

Reply:
cilium_host is a veth device,
```
# ethtool -i cilium_host
driver: veth
version: 1.0
firmware-version: 
expansion-rom-version: 
bus-info: 
supports-statistics: yes
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no
```